templates/CRM/Contact/Form/Task/SaveSearch.tpl

```diff
@@ -25,16 +25,22 @@
 *}
 <div class="crm-form-block crm-block crm-contact-task-createsmartgroup-form-block">
   <h3>{ts}Smart Group{/ts}</h3>
-  {if $qill[0]}
+  <div class="help">
+    <p>{ts}This smart group will stay up-to-date with all contacts who meet the search criteria.{/ts}</p>
+    {if !empty($partiallySelected)}
+      <p>{ts}NOTE: Contacts selected on the search results are not relevant here; all contacts that meet the following criteria will be in the group.{/ts}</p>
```

Comment:
@colemanw should there be a note about if people unsubscribe from the smart group that they won't appear in the group?